src/GoTrueClient.ts

```diff
@@ -2341,25 +2341,38 @@ export default class GoTrueClient {
           return { data: null, error: sessionError }
         }
 
-        const { data, error } = await _request(this.fetch, 'POST', `${this.url}/factors`, {
-          body: {
-            friendly_name: params.friendlyName,
-            factor_type: params.factorType,
-            issuer: params.issuer,
-          },
-          headers: this.headers,
-          jwt: sessionData?.session?.access_token,
-        })
-
-        if (error) {
-          return { data: null, error }
-        }
-
-        if (data?.totp?.qr_code) {
-          data.totp.qr_code = `data:image/svg+xml;utf-8,${data.totp.qr_code}`
+        if (params.factorType === 'phone') {
+          const { data, error } = await _request(this.fetch, 'POST', `${this.url}/factors`, {
+            body: {
+              friendly_name: params.friendlyName,
+              factor_type: params.factorType,
+              phone_number: params.phoneNumber,
+            },
+            headers: this.headers,
+            jwt: sessionData?.session?.access_token,
+          })
+          if (error) {
+            return { data: null, error }
+          }
+          return { data, error: null }
+        } else {
+          const { data, error } = await _request(this.fetch, 'POST', `${this.url}/factors`, {
+            body: {
+              friendly_name: params.friendlyName,
+              factor_type: params.factorType,
+              issuer: params.issuer,
+            },
+            headers: this.headers,
+            jwt: sessionData?.session?.access_token,
+          })
+          if (error) {
+            return { data: null, error }
+          }
+          if (data?.totp?.qr_code) {
+            data.totp.qr_code = `data:image/svg+xml;utf-8,${data.totp.qr_code}`
+          }
```

Comment:
Isn't it easier to have one `_request(this.fetch, ...` call with a single body that's conditionally created above?

Reply:
Yup that was considered but I figured this would better show that the changes are standalone until we lift the feature flag

Reply:
Updated